### PR TITLE
qemu 6.1.0

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by branchvincent** [comment]
```
ERROR: You need at least GCC v7.5 or Clang v6.0 (or XCode Clang v10.0)
```

File: Formula/qemu.rb [hunk 1]
```diff
@@ -95,7 +92,6 @@ def install
     assert_match expected, shell_output("#{bin}/qemu-system-mips64 --version")
     assert_match expected, shell_output("#{bin}/qemu-system-mips64el --version")
     assert_match expected, shell_output("#{bin}/qemu-system-mipsel --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-moxie --version")
```
Comment (branchvincent) [inline Formula/qemu.rb:98]: https://wiki.qemu.org/ChangeLog/6.1#Incompatible_changes
> The 'moxie' target has been removed without replacement

**Comment by branchvincent** [comment]
```
qemu:
  * Binaries built for a non-native architecture were installed into qemu's prefix.
    The offending files are:
      /home/linuxbrew/.linuxbrew/Cellar/qemu/6.1.0/share/qemu/opensbi-riscv32-generic-fw_dynamic.elf	(dunno)
      /home/linuxbrew/.linuxbrew/Cellar/qemu/6.1.0/share/qemu/opensbi-riscv64-generic-fw_dynamic.elf	(dunno)
      /home/linuxbrew/.linuxbrew/Cellar/qemu/6.1.0/share/qemu/palcode-clipper	(dunno)
```

**Review by carlocab — APPROVED** [review]
Non-native binaries seem expected given what this does. We can add this to the [mismatched binary allowlist](https://github.com/Homebrew/homebrew-core/blob/efcea9be43a7ea1e08deec988007747879d2172c/audit_exceptions/mismatched_binary_allowlist.json). Possibly in a separate PR, to avoid having to waste CI time re-running this.

File: Formula/qemu.rb [hunk 2]
```diff
@@ -42,6 +33,12 @@ class Qemu < Formula
   depends_on "snappy"
   depends_on "vde"
 
+  on_linux do
+    depends_on "gcc" => :build
```
Comment (carlocab) [inline Formula/qemu.rb:37]: We should check that this doesn't have a runtime dependency on `gcc`. See https://github.com/Homebrew/homebrew-core/pull/83628#discussion_r693326619

**Pull request closed without merging**